include/exadg/operators/inverse_mass_operator.h

```diff
@@ -43,31 +58,76 @@ class InverseMassOperator
 
   // use a template parameter of -1 to select the precompiled version of this operator
   typedef dealii::MatrixFreeOperators::CellwiseInverseMassMatrix<dim, -1, n_components, Number>
-    CellwiseInverseMass;
+    ExplicitMatrixFreeInverseMass;
 
   typedef std::pair<unsigned int, unsigned int> Range;
 
 public:
-  InverseMassOperator() : matrix_free(nullptr), dof_index(0), quad_index(0)
+  InverseMassOperator()
+    : matrix_free(nullptr),
+      dof_index(0),
+      quad_index(0),
+      explicit_matrix_free_inverse_mass_available(false)
   {
   }
 
   void
   initialize(dealii::MatrixFree<dim, Number> const & matrix_free_in,
-             unsigned int const                      dof_index_in,
-             unsigned int const                      quad_index_in)
+             InverseMassOperatorData const           inverse_mass_operator_data)
   {
     this->matrix_free = &matrix_free_in;
-    dof_index         = dof_index_in;
-    quad_index        = quad_index_in;
+    dof_index         = inverse_mass_operator_data.dof_index;
+    quad_index        = inverse_mass_operator_data.quad_index;
```

Comment:
I am not sure if we should store both dof_index and quad_index as separate element variables in this class, while we could also store the inverse mass operator data only as an element variable.

Reply:
If we store the inverse mass operator data in this class, we store the bool variable  `implement_block_diagonal_preconditioner_matrix_free` and the solver data `solver_data_block_diagonal` as well, although they are only needed in the initialize function. Whereas, dof_index and quad_index are continuously needed in the cell_loop function.